src/content/en/updates/2019/07/image-support-for-async-clipboard.md

```diff
@@ -0,0 +1,311 @@
+{# wf_tags: capabilities,chrome76,cutandcopy,execcommand,input,clipboard #}
+{# wf_featured_snippet: Chrome 76 adds expands the functionality of the Async Clipboard API to add support for png images. Copying and pasting images to the clipboard has never been easier. #}
+{# wf_blink_components: Blink>DataTransfer #}
+
+# Image Support for the Async Clipboard API {: .page-title }
+
+{% include "web/_shared/contributors/thomassteiner.html" %}
+
+In Chrome 66, we shipped the [text portion](/web/updates/2018/03/clipboardapi)
+of the Asynchronous [Clipboard API](https://w3c.github.io/clipboard-apis/).
+Now in Chrome 76, adding support for images to the Asynchronous
+Clipboard API, making it easy to programmatically copy and paste
+`image/png` images.
+
+<aside class="caution">
+  <b>Note:</b>
+  At the time of writing, only <code>image/png</code> files are supported.
+  Support for other images and file formats will be added in the future.
+</aside>
+
+Before we dive in, let’s take a brief look at how the Asynchronous Clipboard
+API works. If you remember the details, skip ahead to the
+[image section](#images).
+
+## Recap of the Asynchronous Clipboard API {: #recap }
+
+### Copy: Writing text to the clipboard {: #copy-text }
+
+Text can be copied to the clipboard by calling `navigator.clipboard.writeText()`.
+Since this API is asynchronous, the `writeText()` function returns a Promise
+that will be resolved or rejected depending on whether the passed text is
+copied successfully:
+
+```js
+async function copyPageURL() {
+  try {
+    await navigator.clipboard.writeText(location.href);
+    console.log('Page URL copied to clipboard');
+  } catch (err) {
+    console.error('Failed to copy: ', err);
+  }
+}
+```
+
+### Paste: Reading text from the clipboard {: #reading-text }
+
+Much like copy, text can be read from the clipboard by calling
+`navigator.clipboard.readText()` and waiting for the returned Promise to
+resolve with the text:
+
+```js
+async function getClipboardText() {
+  try {
+    const text = await navigator.clipboard.readText();
+    console.log('Clipboard contents: ', text);
+  } catch (err) {
+    console.error('Failed to read clipboard contents: ', err);
+  }
+}
+```
+
+### Handling paste events
+
+Paste events can be handled by listening for the (surprise) `paste` event.
+It works nicely with the new asynchronous methods for reading clipboard text:
+
+```js
+document.addEventListener('paste', async (e) => {
+  e.preventDefault();
+  try {
+    const text = await navigator.clipboard.readText();
+    console.log('Pasted text: ', text);
+  } catch (err) {
+    console.error('Failed to read clipboard contents: ', err);
+  }
+});
+```
+
+### Security and permissions {: #security-permission }
+
+The `navigator.clipboard` API is only supported for pages served over HTTPS,
+and to help prevent abuse, clipboard access is only allowed when a page is
+the active tab. Pages in active tabs can write to the clipboard without
```

Comment:
technically, active frames, but tabs is probably easier to understand and may be close enough